services/movie_session.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession, Movie, CinemaHall
+
+
+def create_movie_session(
+        movie_show_time: str,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie=Movie.objects.get(pk=movie_id),
+        cinema_hall=CinemaHall.objects.get(pk=cinema_hall_id)
+    )
+
+
+def get_movies_sessions(session_date: Optional[str] = None) -> QuerySet:
+    if session_date:
+        return MovieSession.objects.filter(show_time__date=session_date)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(pk=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: Optional[str] = None,
+        movie_id: Optional[int] = None,
+        cinema_hall_id: Optional[int] = None
+) -> MovieSession:
+    session = MovieSession.objects.get(pk=session_id)
+
+    if show_time:
+        session.show_time = show_time
+    if movie_id:
+        session.movie = Movie.objects.get(pk=movie_id)
+    if cinema_hall_id:
+        session.cinema_hall = CinemaHall.objects.get(pk=cinema_hall_id)
+
+    session.save()
```

Comment:
Can you refactor this function so that you will not need to get movie session if none of the update parameters has been passed?